### ISSUE TITLE
Add Notebook (.ipynb file) to Launcher

### ISSUE THREAD

**Issue by IrtazaS** [jupyterlab/jupyterlab#5031]
Hi guys,

I am trying to find a way to modify the jupyterlab launcher. For example adding a notebook to the jupyter lab launcher extension?
Sorry if I am asking this question in the wrong section. I hope somebody could help me with this.

**Comment by jasongrout** [comment]
Can you give us a scenario that you would like to work? I'm not sure what you mean by "adding a notebook to the jupyter lab launcher extension".

**Comment by jasongrout** [comment]
Do you mean you want to add a new item in the Launcher tab in JupyterLab? Do you want to add a new icon for a new kernel?

**Comment by IrtazaS** [comment]
yes, I would like to add a new item in the Launcher tab but instead of a kernel I would like to add a specific notebook or a file as a shortcut (eg. a .ipynb file from the file explorer). So it can be opened from the launcher tab just like from the file explorer.

**Comment by jasongrout** [comment]
You can write a new JupyterLab extension to add things to the launcher. For example, here is the code in the terminal extension to add its icon to the launcher: https://github.com/jupyterlab/jupyterlab/blob/0505196b8ec8324e70682af48a42e39c220211fb/packages/terminal-extension/src/index.ts#L119-L126

**Comment by jasongrout** [comment]
I wonder if it makes sense to add a "Favorites" tab that you can add specific files to, similar to how windows or some other file managers have.

**Comment by IrtazaS** [comment]
Thanks Jason! I'll give it a try.
I could probably get the notebook file from the file explorer and add it to the launcher with the extension?

A Favorites tab will definitely be helpful. I am creating a data analysis app for students with notebook. The notebook contains some controls (ipywidgets) and some nb extensions (run all cells, hide code cells etc.) which help students to get started with the "notebook application" without getting into the jupyterlab interface and the ipython code. That's why I wanted to add the notebook on the Launcher Tab so the students don't have to look for it in the file explorer.

**Comment by Fronkan** [comment]
Another use for something at least similar to this is templates. E.g. when doing data processing I have template notebooks I copy into the workspace. It would be realy useful to save these as templates in the launcher and then just instantiate a new one into the current working folder when pressing the button in the launcher.
